platform/darwin/src/MGLBackgroundStyleLayer.mm

```diff
@@ -13,14 +13,25 @@ @interface MGLBackgroundStyleLayer ()
 @property (nonatomic) mbgl::style::BackgroundLayer *layer;
 @property (nonatomic, readwrite) NSString *layerIdentifier;
 @property (nonatomic, readwrite) NSString *sourceIdentifier;
+@property (nonatomic, readwrite) NSString *sourceLayerIdentifier;
 
 @end
 
 @implementation MGLBackgroundStyleLayer
 
 @synthesize mapView;
 
-- (instancetype)initWithLayerIdentifier:(NSString *)layerIdentifier sourceIdentifier:(NSString *)sourceIdentifier {
+- (instancetype)initWithLayerIdentifier:(NSString *)layerIdentifier
+{
+    if (self = [super init]) {
+        _layerIdentifier = layerIdentifier;
+        _layer = new mbgl::style::BackgroundLayer(layerIdentifier.UTF8String);
+    }
+    return self;
+}
+
+- (instancetype)initWithLayerIdentifier:(NSString *)layerIdentifier sourceIdentifier:(NSString *)sourceIdentifier
```

Comment:
Let's take this opportunity to replace the sourceIdentifier parameter with a source parameter that takes an MGLSource.